### PR TITLE
Initial test class created for StarGenerator

### PR DESCRIPTION
**Pull request by ColinLouis95**
StarGeneratorTest.java created with three methods for testing the StarGenerator class

### CODE REVIEW

**Comment by jody** [comment]
When making a pull request, please include evidence that the changes have been reviewed and validated. This could include a transcript of the build and tests being run. and may include comments made by a reviewer (someone who wasn't involved in the development itself).

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        ArrayList<ArrayList<Location>> sp = StarGenerator.generatePattern(3);
+        assertFalse(sp.isEmpty());
+    }
+
+    /** Verifies that the Star Generator method is not randomly generating a pattern */
+    @Test
+    public void StarGeneratorEqualTest(){
+        ArrayList<ArrayList<Location>> sp = StarGenerator.generatePattern(3);
+        ArrayList<ArrayList<Location>> sp2 = StarGenerator.generatePattern(3);
+        assertEquals(sp, sp2);
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:27]: Why do sp and sp2 need to be equal to each other? Is there a specification associated with StarGenerator that indicates that order in the returned list is significant?

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    
+    /** Verifies that the intial location start point for the pattern is 0,0 */
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:30]: Is there a requirement that the first item in the returned collection (list) must be at coordinates 0,0?
Why isn't it sufficient for 0,0 to be just one of the locations in the returned collection?

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Tests for objects of type Location in Star Generator.
+ *
+ * @authors  Colin K, Jesus AE, Thomas M, Robert B
+ * @version 20231012.1
+ */
+
+class StarGeneratorTest{
+
+    /** Verifies that locations inside pattern generated are not empty */
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:15]: What is the defined behavior for StarGenerator with a parameter of 0?
Is the result empty? null? throws and exception?

**Review by jody — COMMENTED** [review]
I'm unsure whether these tests are appropriate to the (admittedly vague) specifications of StarGenerator. Specific comments made on associated lines.

File: src/metrocs/redistricting/StarGeneratorTest.java [hunk 4]
```diff
@@ -12,27 +12,34 @@
...
-    /* Verifies that locations inside pattern generated are not empty */
+    /** Verifies that locations inside pattern generated are not empty */
     @Test
     public void StarGeneratorEmptyTest(){
         ArrayList<ArrayList<Location>> sp = StarGenerator.generatePattern(3);
         assertFalse(sp.isEmpty());
     }
 
-    /* Verifies that the Star Generator method is not randomly generating a pattern */
+    /** Verifies that the Star Generator method is not randomly generating a pattern */
```
Comment (jody) [inline src/metrocs/redistricting/StarGeneratorTest.java:22]: Is there a requirement that the patterns returned by StarGenerator are presented in a specific order? The inherited "equals" method of ArrayList (from AbstractList) requires both lists to have the same elements in the same order.

**Comment by jody** [comment]
Any progress? This PR seems stale. Do you want to close it?

**Comment by rjbeer** [comment]
@jody  The test class has been updated and is ready to be reviewed.

**Comment by jody** [comment]
Please squash commits so I am only seeing a single commit. Thanks!

**Review by jody — CHANGES_REQUESTED** [review]
Please address in-line feedback on individual test case documentation.

Please check that the code matches the project's conventions.
You can use the command `ant checkstyle` and then state it's results in a comment to this PR.

Also please include a transcript of running the product's test suite (ensuring no regression and demonstrating the functionality of the new test class).
You can use the command: `ant test`

**Pull request closed without merging**